src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs

```diff
@@ -399,14 +399,24 @@ public static string GetMSBuildFilePathForPackageReferenceStyleProject(PackageSp
             return Path.Combine(project.RestoreMetadata.OutputPath, $"{projFileName}.nuget.g{extension}");
         }
 
+        /// <summary>
+        /// Generates MSBuild .props/.targets file contents from packages just before completing a restore operation
+        /// </summary>
+        /// <param name="project">Project/package specification</param>
+        /// <param name="assetsFile">project.assets.json model object</param>
+        /// <param name="targetGraphs"></param>
+        /// <param name="repositories">NuGet package sources</param>
+        /// <param name="request">Restore request model</param>
+        /// <param name="assetsFilePath">File Path from loaded <paramref name="assetsFile"/></param>
+        /// <param name="restoreSuccess">Indicates if packages were restored successfully</param>
+        /// <returns>A collection of .props/.targets file contents generated from restored packages</returns>
         public static List<MSBuildOutputFile> GetMSBuildOutputFiles(PackageSpec project,
```

Comment:
Similar logic as some of the other ones. We can probably make the whole thing internal.

Reply:
OK. For now, I made this method internal.